### PR TITLE
Feature/#168 utility conversions

### LINKED ISSUE #168: Utility Conversions
## Feature description

Utility Classes required for Dark Mode. Given a colour, we should be able to identify the dark mode colour and font colour.

Note:
We currently have a color utility and Pau is adding to that

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Acceptance criteria

1. Create a utility class interface (for existing functions)
   1.  Existing functions should be inheritied
   2. Add new functions to the class
2. Create a class to convert colour formats  
3. Create a class to calculate ratios between colours and set colours depending on ratios
4. If there is a dark colour and a dark font, the font should invert to a lighter colour
5. Add unit tests for all functions

## Implementation brief

* 

## QA testing instructions
- Test customizer with different color themes are working as expected. 
- In customizer, test custom theme has warning for color contrast.

Example: 
image

## Demo

* 

## Changelog entry

*

### PR DESCRIPTION
**Pull request by pbearne**
## Summary
Removed all color functions in color-utils.js not in use
Removed chroma-js library
Added tinycolor2 library
Added tests for the color-utils.js

Fixes #168

## Checklist

- [x] My pull request is addressing an [open issue](https://github.com/material-components/material-design-for-wordpress/issues) (please create one otherwise).
- [x] My code is tested and passes existing [tests](https://github.com/material-components/material-design-for-wordpress/blob/main/contributing/engineering.md#tests).
- [x] My code follows the [Contributing Guidelines](https://github.com/material-components/material-design-for-wordpress/blob/main/contributing.md) (updates are often made to the guidelines, check it out periodically).

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -86,7 +86,6 @@
     "babel-plugin-transform-require-context": "0.1.1",
     "bn.js": "5.1.3",
     "browserslist": "4.8.6",
-    "chroma-js": "2.1.0",
```
Comment (PatelUtkarsh) [inline package.json:89]: @pbearne Chroma JS is being used in `plugin/assets/src/customizer/components/material-library/index.js` Should we update the reference for it?

**Comment by emeaguiar** [comment]
@googlebot I consent.

**Comment by PatelUtkarsh** [comment]
@googlebot I consent.

**Review by PatelUtkarsh — COMMENTED** [review]
Hello @pbearne, PR looks good overall.

Current scheme is not visible after checking out this PR which is available in develop branch: 
image

**Comment by pbearne** [comment]
@googlebot I consent.

**Review by PatelUtkarsh — APPROVED** [review]
LGTM!

**Pull request merged**